lib/utils/validateUniqueQuestionIds.ts

```diff
@@ -0,0 +1,24 @@
+import { FormElement } from "@gcforms/types";
+
+export const isUniqueQuestionId = (
+  elements: FormElement[],
+  questionId: string,
+  currentItem: FormElement
+) => {
```

Comment:
I see the diff here checking for the current ID but should we have a small util for 

```
 .flatMap((element: FormElement) => [element, ...(element.properties.subElements || [])])
    .map((element: FormElement) => element.properties?.questionId)
    .filter(Boolean);
```

Thinking if there's ever a bug we might fix one and not the other.